### PR TITLE
Bugfix FXIOS-9172 [Focus] www.x.com doesn't work on Focus after updating to v126.1 (41979)

### PR DESCRIPTION
**Pull request by razvanlitianu**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9172)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/20320)

## :bulb: Description

Current implementation does not allow the list upgrade so content rules are not updated. Added a similar check to compare if the hash has changed.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: focus-ios/Blockzilla/Utilities/LocalContentBlocker.swift [hunk 1]
```diff
@@ -4,12 +4,111 @@
...
+        removeOldListsByHashFromStore { [weak self] in
+            self?.removeOldListsByNameFromStore {
+                self?.getBlockLists { list in
+                    self?.handler?(list)
+                }
+            }
+        }
+    }
+
+    func removeOldListsByNameFromStore(completion: @escaping () -> Void) {
```
Comment (nbhasin2) [inline focus-ios/Blockzilla/Utilities/LocalContentBlocker.swift:26]: nit: private?

File: focus-ios/Blockzilla/Utilities/LocalContentBlocker.swift [hunk 2]
```diff
@@ -4,12 +4,111 @@
...
+        var noMatchingIdentifierFoundForRule = false
+
+        ruleStore?.getAvailableContentRuleListIdentifiers { available in
+            guard let available = available else {
+                completion()
+                return
+            }
```
Comment (nbhasin2) [inline focus-ios/Blockzilla/Utilities/LocalContentBlocker.swift:33]: @razvanlitianu Is this something we want to log or add a error here 🤔

File: focus-ios/Blockzilla/Utilities/LocalContentBlocker.swift [hunk 3]
```diff
@@ -4,12 +4,111 @@
...
+        if hasBlockerFileChanged() {
+            removeAllRulesInStore {
+                completion()
+            }
+        } else {
+            completion()
+        }
+    }
+
+    func removeAllRulesInStore(completion: @escaping () -> Void) {
```
Comment (nbhasin2) [inline focus-ios/Blockzilla/Utilities/LocalContentBlocker.swift:65]: is this the same from Firefox iOS ? 🤔

**Review by nbhasin2 — APPROVED** [review]
some nits, but seems fine as this is the same as FxIOS

File: focus-ios/Blockzilla/Utilities/LocalContentBlocker.swift (hunk 3 above)
Comment (razvanlitianu) [reply focus-ios/Blockzilla/Utilities/LocalContentBlocker.swift]: yes

File: focus-ios/Blockzilla/Utilities/LocalContentBlocker.swift (hunk 2 above)
Comment (razvanlitianu) [reply focus-ios/Blockzilla/Utilities/LocalContentBlocker.swift:33]: The intent seems to be an early exit kind of "fire and forget".

**Comment by razvanlitianu** [comment]
@Mergifyio backport release/v128

**Pull request merged**